src/app/credExplorer/markovChain.js

```diff
@@ -26,3 +26,42 @@ export type SparseMarkovChain = $ReadOnlyArray<{|
   +neighbor: Uint32Array,
   +weight: Float64Array,
 |}>;
+
+export function sparseMarkovChainFromTransitionMatrix(
+  matrix: $ReadOnlyArray<$ReadOnlyArray<number>>
+): SparseMarkovChain {
+  const n = matrix.length;
+  matrix.forEach((row, i) => {
+    if (row.length !== n) {
+      throw new Error(
+        `expected rows to have length ${n}, but row ${i} has ${row.length}`
+      );
+    }
+  });
+  matrix.forEach((row, i) => {
+    row.forEach((value, j) => {
+      if (isNaN(value) || !isFinite(value) || value < 0) {
+        throw new Error(
+          `expected positive real entries, but [${i}][${j}] is ${value}`
```

Comment:
nit: non-negative real entries

Reply:
Absolutely correct (and, I would say, not a nit). Thanks.